File: 094.binary_tree_inorder_traversal/binary_tree_inorder_traversal.cpp
class Solution
{
	public:
		vector<int> inorderTraversal(TreeNode *root)
		{
			vector<int> res;
			stack<const TreeNode *> s;
			TreeNode *p = root;
			while(p != NULL || !s.empty())
			{
				while(p != NULL)
				{
					s.push(p->val);
					p = p->left;
				}
				if(!s.empty())
				{
					p = s.top();
					res.push(p->val);
					s.pop();
					p = p->right;
				}
			}
			return res;
		}
};

File: 094.binary_tree_inorder_traversal/binary_tree_inorder_traversal.py
class Solution(object):
    def inorderTraversal(self, root):
        stack = []
        ret = []
        temp = root
        while temp or stack:
            while temp:
                stack.append(temp)
                temp = temp.left
            if stack:
                temp = stack.pop()
                ret.append(temp.val)
                temp = temp.right
        return ret
